floris/simulation/farm.py

```diff
@@ -72,20 +98,15 @@ def check_y(self, instance: attrs.Attribute, value: Any) -> None:
 
     @turbine_type.validator
     def check_turbine_type(self, instance: attrs.Attribute, value: Any) -> None:
-        if len(value) != len(self.layout_x):
-            if len(value) == 1:
-                value = self.turbine_type * len(self.layout_x)
-            else:
-                raise ValueError("turbine_type must have the same number of entries as layout_x/layout_y or have a single turbine_type value.")
+        if len(value) != self.n_turbines and len(value) != 1:
+            raise ValueError("turbine_type must have the same number of entries as layout_x/layout_y or have a single turbine_type value.")
 
-        self.turbine_definitions = copy.deepcopy(value)
-        for i, val in enumerate(value):
-            if type(val) is str:
-                _floris_dir = Path(__file__).parent.parent
-                fname = _floris_dir / "turbine_library" / f"{val}.yaml"
+        for val in value:
+            if isinstance(type(val), str):
+                floris_dir = Path(__file__).parent.parent
```

Comment:
Fairly nitpicky, and not necessary, but you can do `Path(__file__).parents[1]` for more easily going up more than one directory.